app/actions/battle_round.py

```diff
@@ -0,0 +1,12 @@
+def battle_round(knights: dict) -> dict:
+
+    knights["Lancelot"].hp -= (knights["Mordred"].power
+                               - knights["Lancelot"].protection)
+    knights["Mordred"].hp -= (knights["Lancelot"].power
+                              - knights["Mordred"].protection)

```

Comment:
Create separate function
def battle(first_knight, second_knight):
and then invoke it no need to duplicate code